packages/react/src/components/Table/TableHeader.tsx

```diff
@@ -11,45 +12,69 @@ interface TableHeaderProps extends ThHTMLAttributes<HTMLTableCellElement> {
   sortDirection?: SortDirection;
   onSort?: () => void;
   className?: string;
+  sortAscendingAnnouncement?: string;
+  sortDescendingAnnoucnement?: string;
 }
 
 const TableHeader = ({
   children,
   sortDirection,
   onSort,
   className,
+  sortAscendingAnnouncement = 'sorted ascending',
+  sortDescendingAnnoucnement = 'sorted descending',
   ...other
-}: TableHeaderProps) => (
-  <th
-    aria-sort={sortDirection}
-    className={classNames('TableHeader', className, {
-      'TableHeader--sort-ascending': sortDirection === 'ascending',
-      'TableHeader--sort-descending': sortDirection === 'descending'
-    })}
-    {...other}
-  >
-    {!!onSort && !!sortDirection ? (
-      <button
-        onClick={onSort}
-        className="TableHeader__sort-button"
-        type="button"
-      >
-        {children}
-        <span aria-hidden="true">
-          {sortDirection === 'none' ? (
-            <Icon type="sort-triangle" />
-          ) : sortDirection === 'ascending' ? (
-            <Icon type="triangle-up" />
-          ) : (
-            <Icon type="triangle-down" />
-          )}
-        </span>
-      </button>
-    ) : (
-      children
-    )}
-  </th>
-);
+}: TableHeaderProps) => {
+  // When the sort direction changes, we want to announce the change in a live region
+  // because changes to the sort value is not widely supported yet
+  // see: https://a11ysupport.io/tech/aria/aria-sort_attribute
+  const [announcement, setAnnouncement] = useState('');
+  useEffect(() => {
+    if (sortDirection !== 'none') {
+      setAnnouncement(
+        sortDirection === 'ascending'
+          ? sortAscendingAnnouncement
+          : sortDescendingAnnoucnement
+      );
+    }
+  }, [sortDirection]);
```

Comment:
I don't think there's any value here in storing this as state:

```suggestion
 const announcement = sortDirection === 'ascending' ? sortAscendingAnnouncement : sortDirection === 'descending' ? sortDescendingAnnouncement : ''; 
```

Reply:
There is, because we don't want to announce the _initial_ state of the sort, only updates.

Reply:
`useEffect` runs on the initial render, so if the sort direction is something other than `none`, so wouldn't it suffer from that same problem?

Reply:
Actually, I don't _think_ it would read it either way. In any case, I think the behavior would be the same without the `useEffect` and useState`